pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/ThreadRuntimeFactory.java

```diff
@@ -41,26 +45,37 @@
     private final String storageServiceUrl;
     private volatile boolean closed;
 
-    public ThreadRuntimeFactory(String threadGroupName,
-                                String pulsarServiceUrl,
-                                String storageServiceUrl)
-            throws Exception {
-        this(
-            threadGroupName,
-            pulsarServiceUrl != null ? PulsarClient.builder().serviceUrl(pulsarServiceUrl).build() : null,
-            storageServiceUrl);
+    public ThreadRuntimeFactory(String threadGroupName, String pulsarServiceUrl, String storageServiceUrl,
+            String clientAuthenticationPlugin, String clientAuthenticationParameters, boolean useTls,
+            boolean tlsAllowInsecureConnection) throws Exception {
+        this(threadGroupName, createPulsarClient(pulsarServiceUrl, clientAuthenticationPlugin,
+                clientAuthenticationParameters, useTls, tlsAllowInsecureConnection), storageServiceUrl);
     }
 
     @VisibleForTesting
-    ThreadRuntimeFactory(String threadGroupName,
-                         PulsarClient pulsarClient,
-                         String storageServiceUrl) {
+    ThreadRuntimeFactory(String threadGroupName, PulsarClient pulsarClient, String storageServiceUrl) {
         this.fnCache = new FunctionCacheManagerImpl();
         this.threadGroup = new ThreadGroup(threadGroupName);
         this.pulsarClient = pulsarClient;
         this.storageServiceUrl = storageServiceUrl;
     }
 
+    private static PulsarClient createPulsarClient(String pulsarServiceUrl, String clientAuthenticationPlugin,
+            String clientAuthenticationParameters, boolean useTls, boolean tlsAllowInsecureConnection)
+            throws PulsarClientException {
+        ClientBuilder clientBuilder = null;
+        if (isNotBlank(pulsarServiceUrl)) {
+            clientBuilder = PulsarClient.builder().serviceUrl(pulsarServiceUrl);
+            if (isNotBlank(clientAuthenticationPlugin) && isNotBlank(clientAuthenticationParameters)) {
+                clientBuilder.authentication(clientAuthenticationPlugin, clientAuthenticationParameters);
+                clientBuilder.enableTls(useTls);
```

Comment:
Enabled tls only if authentication is enabled?

Reply:
fixed it.